wear/src/main/java/io/homeassistant/companion/android/sensors/HealthServicesSensorManager.kt

```diff
@@ -0,0 +1,115 @@
+            "mdi:account",
+            entityCategory = SensorManager.ENTITY_CATEGORY_DIAGNOSTIC,
+            updateType = SensorManager.BasicSensor.UpdateType.INTENT
+        )
+    }
+
+    private lateinit var latestContext: Context
+    private var healthClient: HealthServicesClient? = null
+    private var passiveMonitoringClient: PassiveMonitoringClient? = null
+    private var passiveListenerConfig: PassiveListenerConfig? = null
+    private var callBackRegistered = false
+
+    override fun docsLink(): String {
+        return "https://companion.home-assistant.io/docs/wear-os/#sensors"
+    }
+    override val enabledByDefault: Boolean
+        get() = false
+
+    override val name: Int
+        get() = commonR.string.sensor_name_health_services
+
+    override fun getAvailableSensors(context: Context): List<SensorManager.BasicSensor> {
+        return listOf(userActivityState)
+    }
+
+    override fun requiredPermissions(sensorId: String): Array<String> {
+        return arrayOf(Manifest.permission.ACTIVITY_RECOGNITION)
+    }
+
+    override fun hasSensor(context: Context): Boolean {
+        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.R
+    }
+
+    override fun requestSensorUpdate(context: Context) {
+        latestContext = context
+        updateUserActivityState()
+    }
+
+    private fun updateUserActivityState() {
+        if (!isEnabled(latestContext, userActivityState.id)) {
+            passiveMonitoringClient?.clearPassiveListenerCallbackAsync()
+            callBackRegistered = false
+            return
+        }
+
+        if (healthClient == null) healthClient = HealthServices.getClient(latestContext)
+        if (passiveMonitoringClient == null) passiveMonitoringClient = healthClient?.passiveMonitoringClient
+        passiveListenerConfig = PassiveListenerConfig.builder()
+            .setShouldUserActivityInfoBeRequested(isEnabled(latestContext, userActivityState.id))
+            .build()
+
+        val passiveListenerCallback: PassiveListenerCallback = object : PassiveListenerCallback {
+            override fun onUserActivityInfoReceived(info: UserActivityInfo) {
+                Log.d(TAG, "User activity state: ${info.userActivityState.name}")
+                onSensorUpdated(
+                    latestContext,
+                    userActivityState,
+                    when (info.userActivityState) {
+                        UserActivityState.USER_ACTIVITY_ASLEEP -> "asleep"
+                        UserActivityState.USER_ACTIVITY_PASSIVE -> "passive"
+                        UserActivityState.USER_ACTIVITY_EXERCISE -> "exercise"
+                        else -> "unknown"
+                    },
+                    when (info.userActivityState) {
+                        UserActivityState.USER_ACTIVITY_EXERCISE -> "mdi:run"
+                        UserActivityState.USER_ACTIVITY_ASLEEP -> "mdi:sleep"
+                        UserActivityState.USER_ACTIVITY_PASSIVE -> "mdi:walk"

```

Comment:
Walking != passive in my opinion... Suggestion: something that doesn't imply movement like [`mdi:human-handsdown`](https://materialdesignicons.com/icon/human-handsdown)? (The documentation says this is when "[t]he user is awake but is not currently exercising", so it could be correct, but passive implies less.)

Reply:
good suggestion, sometimes its not easy to find these icons lol